### ISSUE TITLE
Report variants with ref mismatches [maf2maf]

### ISSUE THREAD

**Issue by kpjonsson** [mskcc/vcf2maf#38]
It would be convenient to write variants to a separate file if they are skipped by maf2maf due to reference mismatches.

**Comment by ckandoth** [comment]
Note that [this recent commit](https://github.com/mskcc/vcf2maf/commit/004835ac30afee1fb372e34add6795f81fbec770) should make reference mismatches less likely. I'll still add this feature, but it's gotta go on the backburner.

**Comment by ckandoth** [comment]
Since `--check_ref` is implemented in VEP itself, there are 2 ways of doing this:
1. Implement a reference allele check in vcf2maf/maf2maf before running VEP, so we can control what it does.
2. Have vcf2maf/maf2maf parse mismatched ref warnings generated by VEP, and figure out which variants those belong to.

Option 2 is way too cumbersome, and likely not feasible because those VEP warnings don't report genomic loci. Option 1 is possible as a check that will be disabled by default, because it considerably slows down MAF/VCF parsing, as each allele is checked against the reference FASTA.

I won't close this issue, since it's still a good feature to have, but it's gotta remain in the backburner for now.

**Comment by ckandoth** [comment]
c02689fb4ceaeb485d374024e336e08bbf8caa3c implements option 1 for vcf2maf. `samtools faidx` on all variant loci at the same time, runs really quick. So this pre-annotation reference allele check is not optional. It's not yet implemented in maf2maf/maf2vcf because there's already code to handle the crappiness allowed in an input file. And I need to be careful how it's modified for additional ref checks.

**Comment by ckandoth** [comment]
9d1ca04 implements reference allele checks in `maf2vcf`, which is used by `maf2maf`. So use `maf2maf` with `--tmp-dir` where a separate file will be created if at least 1 reference allele mismatch is detected. The variants dumped into this file are not sent to VEP for annotation.

@kpjonsson can you test this out and report back? Will keep issue open till then. Here's how to use this on luna:
```
cmo_maf2maf --version develop --input-maf test.maf --output-maf test.vep.maf --tmp-dir anno_files
```

**Comment by kpjonsson** [comment]
@ckandoth For your troubleshooting purposes: `/ifs/res/taylorlab/jonssonp/FMI_solid_tumor_data_release/tmp`

I ran `cmo_maf2maf --input-maf test.maf --output-maf test.vep.maf --tmp-dir vep_test/ --ncbi-build GRCh37` and it ran until completion but for the concatenation of the individual per-sample MAFs into a large MAF.

**Comment by ckandoth** [comment]
Thanks much for this test MAF. It has 2 calls where `REF==ALT`, which caused an infinite loop [here](https://github.com/mskcc/vcf2maf/commit/768935687aa5641d92a408d7ace65d11f94caf31#diff-d2810e3979b0592c29dd3056f6535dacR261). Now fixed in 0b9fc73.

**Comment by ckandoth** [comment]
Closing issue. As originally described, `maf2maf` will now print the following when the first reference mismatch is detected:

`WARNING: Reference allele mismatches found. Storing them here for debugging: tmp_dir/test.skipped.tsv`
where `tmp_dir` is the directory specified with `--tmp-dir`, and `test` is the name of the `--input-maf`.

One caveat with deletions/substitutions that are >1 bp long, is that only the first bp of the reference allele is checked. This maintains the non-dependence on `Stop_Position`, to find the number of alleles to pull from the reference FASTA. Alternatively, I could use the length of the reported reference allele to figure out how many bps to pull from the FASTA. Will make a separate issue to handle this.
